src/apps/ocioconvert/main.cpp

```diff
@@ -513,22 +551,40 @@ int main(int argc, const char **argv)
             oglApp->setShader(shaderDesc);
             oglApp->reshape(imgwidth, imgheight);
             oglApp->redisplay();
-            oglApp->readImage((float *)img.getBuffer());
+            oglApp->readImage((float *)imgInput.getBuffer());
         }
         else
 #endif // OCIO_GPU_ENABLED
         {
-            const OCIO::BitDepth bitDepth = OCIO::GetBitDepth(spec);
+            // Copy specInput into specOutput.
+            specOutput = specInput;
+            if (specInput.format == OIIO::TypeDesc::UINT16 || specInput.format == OIIO::TypeDesc::FLOAT) 
```

Comment:
[Question] Why are we promoting UINT16 to FLOAT and UINT8 to HALF? Would it be worth to add a comment in case that is indeed the right thing to do?

Reply:
It was the result of a discussion with @doug-walker. I'll add a comment based on that.

Reply:
Thanks, note that in the related PR #1637 all the Image IO code is gone from the apps into an ImageIO class so this code may need porting later on.